test/object-store/realm.cpp

```diff
@@ -993,6 +993,8 @@ TEST_CASE("Get Realm using Async Open", "[asyncOpen]") {
             realm->read_group().get_table("class_object")->create_object_with_primary_key(0);
             realm->commit_transaction();
             wait_for_upload(*realm);
+            // Initialize a realm with config before locking it
+            auto origin = Realm::get_shared_realm(config);
```

Comment:
What does this do?

Reply:
This forces the empty realm file to be initialized before holding the write lock later on for the test. The test was getting stuck in the migration store when it was initializing the schema versions table, since it needs to be able to write the metadata to the realm file.

Reply:
I don't super love that the MigrationStore is going to have these kinds of side-effects where realms that are not going ever going to do a migration have to do special initialization order stuff to make progress. And I still don't quite understand why we have to do this - we're creating the DB via `DB::create` which shouldn't initialize or touch the sync client in any way. Don't we want to block creating the migration store tables until we cancel the write?

Reply:
I added a suggestion in migration_store.cpp for how to prevent this kind of side-effect. Let me know what you think.